common/djangoapps/student/tests/test_enrollment.py

```diff
@@ -30,7 +31,7 @@
 @ddt.ddt
 @patch.dict('django.conf.settings.FEATURES', {'ENABLE_SPECIAL_EXAMS': True})
 @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
-class EnrollmentTest(UrlResetMixin, SharedModuleStoreTestCase):
+class EnrollmentTest(UrlResetMixin, SharedModuleStoreTestCase, OpenEdxEventsTestMixin):
```

Comment:
Since the default behavior for the `OpenEdxEventsTestMixin` is that it will not send out events unless specified in `ENABLED_OPENEDX_EVENTS` I suggest we make it explicit for the first tests.
By this I mean, inherit from the mixin and also put a `ENABLED_OPENEDX_EVENTS = []` statement.

Reply:
This way anyone looking at this as usage example will learn about the existence of the ENABLED_OPENEDX_EVENTS key setting

Reply:
Great, thanks for the suggestion!

Reply:
Where is the corresponding integration test for when `ENABLED_OPENEDX_EVENTS` is _not_ an empty list?

Reply:
Good point, we can make it explicit for the `test_enrollment_created_event_emitted` test below.

Also any TestCase that does not inherit `OpenEdxEventsTestMixin` will act as regular code not bound by test restrictions and send the events normally.

Reply:
Thanks for the suggestion @nasthagiri! I changed the events tests -e.g. [EnrollmentEventsTest](https://github.com/edx/edx-platform/blob/28692e37b6222dc78763732f7d4af55af4183f6b/common/djangoapps/student/tests/test_events.py#L198)-, so they correspond to the integration of each event to the login, registration, and enrollment process.